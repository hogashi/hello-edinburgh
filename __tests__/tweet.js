import * as React from 'react';
import renderer from 'react-test-renderer';
import { act } from 'react-dom/test-utils';

import Tweet from '../src/components/tweet';

const getUrls = (short_url) => {
  return {
    short_url,
    expanded_url: `${short_url}ToBeExpended`,
    display_url: `${short_url}ToBeDisplayed`,
    actual_url: `${short_url}ToBeActual`,
  };
};

const getTweetBase = () => {
  return {
    id: 0,
    tweet_url: 'tweet.url',
    created_at: Math.floor(new Date('2019-03-17T18:47:20').getTime() / 1000),
    user: {
      icon: 'user.icon',
      name: 'user.name',
      screen_name: 'user.screenName',
    },
    colorIndex: 0,
  };
};

const getTweet = () => {
  return {
    ...getTweetBase(),
    text: 'default text',
    urls: [],
    media_urls: [],
    time: 0,
  };
};

let container;

beforeEach(() => {
  container = global.document.createElement('div');
  global.document.body.appendChild(container);
});

afterEach(() => {
  global.document.body.removeChild(container);
  container = null;
});

describe('Tweet', () => {
  it('render normal tweet', () => {
    const tweet = getTweet();
    let component;
    act(() => {
      component = renderer.create(
        <Tweet {...tweet} />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render retweet', () => {
    const tweet = getTweet();
    tweet.retweeter = getTweetBase();
    let component;
    act(() => {
      component = renderer.create(
        <Tweet {...tweet} />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render tweet with urls', () => {
    const tweet = getTweet();
    tweet.text = 'hoge url1 fuga url2 url3';
    tweet.urls = [
      getUrls('url1'),
      getUrls('url2'),
      getUrls('url3'),
    ];
    let component;
    act(() => {
      component = renderer.create(
        <Tweet {...tweet} />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render tweet with media', () => {
    const tweet = getTweet();
    tweet.text = 'hoge url1 fuga';
    tweet.media_urls = [
      getUrls('url1'),
      getUrls('url2'),
      getUrls('url3'),
    ];
    let component;
    act(() => {
      component = renderer.create(
        <Tweet {...tweet} />
      );
    });

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
