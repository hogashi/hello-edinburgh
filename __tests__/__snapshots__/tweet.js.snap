// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweet render normal tweet 1`] = `
<div
  className="tweet"
  data-id={0}
>
  <div
    className="icon"
  >
    <img
      src="user.icon"
    />
  </div>
  <div
    className="contents"
  >
    <div
      className="user"
    >
      <span
        className="name"
      >
        user.name
      </span>
      <span
        className="screenName"
      >
        @
        user.screenName
      </span>
      <span
        className="createdAt"
      >
        <a
          href="tweet.url"
          target="_blank"
        >
          2019/02/17-18:47:20
        </a>
      </span>
    </div>
    <div
      className="text"
    >
      default text
    </div>
    <div
      className="media"
    />
  </div>
</div>
`;

exports[`Tweet render retweet 1`] = `
<div
  className="tweet"
  data-id={0}
>
  <div
    className="icon"
  >
    <img
      src="user.icon"
    />
  </div>
  <div
    className="contents"
  >
    <div
      className="user"
    >
      <span
        className="name"
      >
        user.name
      </span>
      <span
        className="screenName"
      >
        @
        user.screenName
      </span>
      <span
        className="createdAt"
      >
        <a
          href="tweet.url"
          target="_blank"
        >
          2019/02/17-18:47:20
        </a>
      </span>
    </div>
    <div
      className="text"
    >
      default text
    </div>
    <div
      className="media"
    />
    <div
      className="retweeter"
    >
      <span
        className="rtby"
      >
        RTby
      </span>
      <span
        className="icon"
      >
        <img
          src="user.icon"
        />
      </span>
      <span
        className="name"
      >
        user.name
      </span>
      <span
        className="screenName"
      >
        @
        user.screenName
      </span>
      <span
        className="createdAt"
      >
        <a
          href="tweet.url"
          target="_blank"
        >
          2019/02/17-18:47:20
        </a>
      </span>
    </div>
  </div>
</div>
`;

exports[`Tweet render tweet with media 1`] = `
<div
  className="tweet"
  data-id={0}
>
  <div
    className="icon"
  >
    <img
      src="user.icon"
    />
  </div>
  <div
    className="contents"
  >
    <div
      className="user"
    >
      <span
        className="name"
      >
        user.name
      </span>
      <span
        className="screenName"
      >
        @
        user.screenName
      </span>
      <span
        className="createdAt"
      >
        <a
          href="tweet.url"
          target="_blank"
        >
          2019/02/17-18:47:20
        </a>
      </span>
    </div>
    <div
      className="text"
    >
      hoge 
      <a
        href="url1ToBeExpended"
        target="_blank"
      >
        url1ToBeDisplayed
      </a>
       fuga
    </div>
    <div
      className="media"
    >
      <div
        className="medium"
      >
        <img
          src="url1ToBeActual"
        />
      </div>
      <div
        className="medium"
      >
        <img
          src="url2ToBeActual"
        />
      </div>
      <div
        className="medium"
      >
        <img
          src="url3ToBeActual"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Tweet render tweet with urls 1`] = `
<div
  className="tweet"
  data-id={0}
>
  <div
    className="icon"
  >
    <img
      src="user.icon"
    />
  </div>
  <div
    className="contents"
  >
    <div
      className="user"
    >
      <span
        className="name"
      >
        user.name
      </span>
      <span
        className="screenName"
      >
        @
        user.screenName
      </span>
      <span
        className="createdAt"
      >
        <a
          href="tweet.url"
          target="_blank"
        >
          2019/02/17-18:47:20
        </a>
      </span>
    </div>
    <div
      className="text"
    >
      hoge 
      <a
        href="url1ToBeExpended"
        target="_blank"
      >
        url1ToBeDisplayed
      </a>
       fuga 
      <a
        href="url2ToBeExpended"
        target="_blank"
      >
        url2ToBeDisplayed
      </a>
       
      <a
        href="url3ToBeExpended"
        target="_blank"
      >
        url3ToBeDisplayed
      </a>
      
    </div>
    <div
      className="media"
    />
  </div>
</div>
`;
